### ISSUE TITLE
extension build error: can't be assigned to the parameter type

### ISSUE THREAD

**Issue by mnordine** [dart-lang/sdk#38942]
Dart SDK version: `2.6.0-dev.8.0`
MacOS

```dart
class Thing {
}

extension Animations on Thing {
  void fade({num to = 1}) {}
}

void main() {
  Thing().fade(to: 1);
}
```

Results in `build_web_compilers` error:
```
[SEVERE]build_web_compilers:ddc on web/main.ddc.module (cached): Error compiling dartdevc module:extensions_test|web/main.ddc.js

web/main.dart:11:20: Error: The argument type 'Thing' can't be assigned to the parameter type 'num'.
 - 'Thing' is from 'web/main.dart'.
  Thing().fade(to: 1);
                   ^


[SEVERE]build_web_compilers:entrypoint on web/main.dart (cached): AssetNotFoundException: extensions_test|web/main.ddc.js

[SEVERE] Failed after 353ms
```

**Comment by vsmenon** [comment]
Seeing this from the CFE as of a few days ago:
```
> ~/dart/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dart example.dart 
example.dart:9:20: Error: The argument type 'Thing' can't be assigned to the parameter type 'num'.
 - 'Thing' is from 'example.dart'.
  Thing().fade(to: 1);
                   ^
> ~/dart/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dart --version
Dart VM version: 2.6.0-edge.19e2f034212d7e1e0fc901d0bac9f404066746b2 (Tue Oct 15 17:56:08 2019 +0000) on "macos_x64"
```

fyi @johnniwinther

**Comment by vsmenon** [comment]
Adding to D27.  If an easy fix, we may want to cherry pick to D26.

**Comment by vsmenon** [comment]
This may be a dup of:

https://github.com/dart-lang/sdk/issues/38915

**Comment by johnniwinther** [comment]
This is a duplicate of #38915 which has been fixed.

**Comment by mnordine** [comment]
Still not fixed in `2.6.0-dev.8.1`
